src/ray/rpc/worker/core_worker_client.h

```diff
@@ -217,12 +211,27 @@ class CoreWorkerClient : public std::enable_shared_from_this<CoreWorkerClient>,
       : addr_(address) {
     grpc_client_ = std::make_unique<GrpcClient<CoreWorkerService>>(
         addr_.ip_address(), addr_.port(), client_call_manager);
+
+    retryable_grpc_client_ = RetryableGrpcClient::Create(
+        grpc_client_->Channel(),
+        client_call_manager.GetMainService(),
+        /*max_pending_requests_bytes=*/
+        std::numeric_limits<uint64_t>::max(),
+        /*check_channel_status_interval_milliseconds=*/
+        ::RayConfig::instance()
+            .gcs_client_check_connection_status_interval_milliseconds(),
+        /*server_unavailable_timeout_seconds=*/
+        std::numeric_limits<uint64_t>::max(),
+        /*server_unavailable_timeout_callback=*/
+        []() { RAY_LOG(FATAL) << "Server unavailable should never timeout"; },
```

Comment:
are there other possible handler for server unavailable? if not we can build the FATAL in RetryableGrpcClient

Reply:
Different client needs different policy: FATAL makes sense for gcs_client but doesn't necessarily make sense for other clients like (core_worker_client).

Reply:
so here we are in CoreWorkerClient, shall we update the callback to log-and-remove-core-worker?

Reply:
@rynewang I updated the PR to handle server unavailable for `ReportGeneratorItemReturns`, PTAL.